### PR TITLE
drawer update

### PR DESCRIPTION
**Pull request by killalad**

### CODE REVIEW

File: zScanner/Coordinators/DocumentsCoordinator.swift [hunk 1]
```diff
@@ -69,3 +85,13 @@ extension DocumentsCoordinator: NewDocumentFlowDelegate {
...
     }
 }
+
+// MARK: - MenuFlowDelegate implementation
+extension DocumentsCoordinator: MenuFlowDelegate {
+    func logout() {
+        flowDelegate.logout()
+    }
+    func deleteHistory() {
+        viewModel.deleteHistory()
```
Comment (jakubskorepapoq) [inline zScanner/Coordinators/DocumentsCoordinator.swift]: Don't call it on the viewModel. Keep the viewModel exclusive for showDocumentsListScreen and call
just `database.deleteAll()` here. Also, I have added FolderDatabaseModel, so delete Folders too

Comment (skorepak) [inline zScanner/Coordinators/DocumentsCoordinator.swift]: Don't call it on the viewModel. Keep the viewModel exclusive for showDocumentsListScreen and call
just `database.deleteAll()` here. Also, I have added FolderDatabaseModel, so delete Folders too

File: zScanner/Coordinators/DocumentsCoordinator.swift [hunk 2]
```diff
@@ -69,3 +85,13 @@ extension DocumentsCoordinator: NewDocumentFlowDelegate {
         list.insertNewDocument(document: documentViewModel)
     }
 }
+
+// MARK: - MenuFlowDelegate implementation
+extension DocumentsCoordinator: MenuFlowDelegate {
+    func logout() {
+        flowDelegate.logout()
```
Comment (skorepak) [inline zScanner/Coordinators/DocumentsCoordinator.swift]: call deleteHistory before the login

File: zScanner/Coordinators/DocumentsCoordinator.swift [hunk 3]
```diff
@@ -56,6 +69,9 @@ extension DocumentsCoordinator: DocumentsListCoordinator {
         tracker.track(.documentModeSelected(mode))
         runNewDocumentFlow(with: mode)
     }
+    func openDrawer() {
```
Comment (skorepak) [inline zScanner/Coordinators/DocumentsCoordinator.swift]: Rename to openMenu. It's just better naming :)

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        bView.alpha = 0
+        return bView
+    }()
+    private lazy var width: CGFloat = {
+        let width = window.frame.width * 2 / 3
+        return width
+    }()
+
+    func begin() {
+        window.addSubview(blackView)
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: wrap this code into `func installDrawerScreen()` and call it from begin. 

Also, take a look into SnapKit, We are using it in the rest of the project. Would be nice to use it here too.

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    // MARK: Instance part
+    unowned private let flowDelegate: MenuFlowDelegate
+
+    init(flowDelegate: MenuFlowDelegate, window: UIWindow, navigationController: UINavigationController? = nil) {
+        self.flowDelegate = flowDelegate
+
+        super.init(window: window, navigationController: navigationController)
+
+    }
+    private lazy var viewController: DrawerViewController = {
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: Use more specific naming. like `drawerViewController`

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        return DrawerViewController(coordinator: self)
+    }()
+    
+    private lazy var blackView: UIView = {
+        let bView = UIView()
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: Don't use "cripled" names 😄  use either blackView or view. It's fine to use just "view" because the name will not leave this scope

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        bView.backgroundColor = .black
+        bView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(handleDismiss)))
+        let swipeLeftGesture = UISwipeGestureRecognizer(target: self, action: #selector(handleDismiss))
+        swipeLeftGesture.direction = .left
+        bView.addGestureRecognizer(swipeLeftGesture)
+        bView.translatesAutoresizingMaskIntoConstraints = false
+        bView.alpha = 0
+        return bView
+    }()
+    private lazy var width: CGFloat = {
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: Use better naming, this is not width of the coordinator. Name it like `menuWidth` or something more meaningful

File: zScanner/ViewControllers/Documents/DocumentsListViewController.swift [hunk 8]
```diff
@@ -12,14 +12,15 @@ import RxSwift
 
 protocol DocumentsListCoordinator: BaseCoordinator {
     func createNewDocument(with mode: DocumentMode)
+    func openDrawer()
```
Comment (skorepak) [inline zScanner/ViewControllers/Documents/DocumentsListViewController.swift]: rename to `openMenu()` please 🙌

File: zScanner/ViewControllers/Documents/DocumentsListViewController.swift [hunk 9]
```diff
@@ -81,6 +88,9 @@ class DocumentsListViewController: BaseViewController {
     @objc private func newDocument() {
         showDocumentModePicker()
     }
+    @objc private func hamburgerTap() {
```
Comment (skorepak) [inline zScanner/ViewControllers/Documents/DocumentsListViewController.swift]: rename to openMenu(), call coordinator directly here. no need for another function.

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        case drawerMenuItems[2]:
+            coordinator.showAbout()
+        default:
+            print("bug")
+        }
+        
+    }
+    
+    func hideDrawer() {
+        view.frame = CGRect(x: -view.frame.width, y: 0, width: view.frame.width, height: view.frame.height)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: You are adjusting the viewController position form two places. Coordinator and from here. Select only one. I would go with coordinator, as the vc doesn't need to know about the way it's being presented. 

Also, set directly `view.frame.origin` so you don't need to specify width and height when not changing it. 👍

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+    private lazy var topStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [drawerLogo,drawerTopLabel])
+        stackView.axis = .vertical
+        stackView.alignment = .top
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.spacing = 10
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.spacing = UIStackView.spacingUseSystem
+
+        stackView.addBackground(color: .primary)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: why are you not using `stackView.backgroundColor = .primary` ?

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    
+
+    private func setupView() {
+        view.addSubview(topStackView)
+        view.addSubview(menuStackView)
+        
+        topStackView.layoutMargins = UIEdgeInsets(top: 20, left: 20, bottom: 20, right: 20)
+
+
+        topStackView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: Please use SnapKit for consistency

File: zScanner/Views/Buttons/DrawerMenuButton.swift [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+        super.init(frame: frame)
+        
+        setup()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    open override var isEnabled: Bool{
```
Comment (skorepak) [inline zScanner/Views/Buttons/DrawerMenuButton.swift]: Why do you need isEnabled in menu buttons?

File: zScanner/Views/Buttons/DrawerMenuButton.swift [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+        didSet {
+            alpha = isEnabled ? 1.0 : 0.5
+        }
+    }
+    
+    //MARK: Helpers
+    private func setup() {
+        translatesAutoresizingMaskIntoConstraints = false
```
Comment (skorepak) [inline zScanner/Views/Buttons/DrawerMenuButton.swift]: Don't set translatesAutoresizingMaskIntoConstraints from here. This is logic from where the button is used, not of the button itself.

**Comment by skorepak** [comment]
Your formatting is not consistent. 
Please use one empty line before every function or multiline variable specification. 
Don't use empty lines before end of scope (line ending `}`)
Use single space after every comma

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+        return stackView
+    }()
+
+    private func setupView() {
+        view.addSubview(topView)
+        view.addSubview(menuStackView)
+        
+        // MARK: - Constraints
+        topView.snp.makeConstraints { make in
+            make.top.equalTo(self.view)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: You can make it single line as make.top.left.right.equalToSuperview()

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 16]
```diff
@@ -0,0 +1,138 @@
...
+            make.left.equalTo(self.view)
+            make.right.equalTo(self.view)
+        }
+
+        drawerLogo.snp.makeConstraints { make in
+            make.height.equalTo(102)
+            make.width.equalTo(114)
+            make.top.equalTo(topView).offset(40)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: equalToSuperview()

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 17]
```diff
@@ -0,0 +1,138 @@
...
+        label.text = "zScanner"
+        label.textColor = .white
+        label.font = label.font.withSize(25)
+        return label
+    }()
+    
+    private lazy var topView: UIView = {
+        let topView = UIView()
+        
+        topView.addSubview(drawerLogo)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: please add subviews in the setupView function. Here you are creating components and in the setupView functions you are  laying them up.

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 18]
```diff
@@ -0,0 +1,138 @@
...
+            button.translatesAutoresizingMaskIntoConstraints = false
+            button.addTarget(self, action: #selector(handleMenuTap(_:)), for: .touchUpInside)
+
+            drawerMenu.append(button)
+        }
+        return drawerMenu
+    }
+    
+    private lazy var menuStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: drawerMakeMenuButtons(drawerMenuItems))
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: don't add subviews in here. Move it to the setupVIew() function

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 19]
```diff
@@ -0,0 +1,138 @@
...
+        }
+        
+        menuStackView.snp.makeConstraints { make in
+            make.top.equalTo(topView.snp.bottom).offset(15)
+            make.left.equalTo(view).offset(15)
+            make.right.equalTo(view).offset(15)
+        }
+    }
+    
+    @objc func handleMenuTap(_ sender: UIButton){
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: make the function private

Comment (skorepak) [reply zScanner/ViewControllers/Drawer/DrawerViewController.swift]: space missing at the end of the line

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 20]
```diff
@@ -0,0 +1,138 @@
...
+            make.height.equalTo(102)
+            make.width.equalTo(114)
+            make.top.equalTo(topView).offset(40)
+            make.left.equalTo(topView).offset(20)
+        }
+        
+        drawerTopLabel.snp.makeConstraints { make in
+            make.top.equalTo(drawerLogo.snp.bottom).offset(20)
+            make.left.equalTo(topView).offset(20)
+            make.right.equalTo(topView).offset(20)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: I believe there should be -20, but instead you can do make.edges.equalToSuperview().**inset**(20)

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 21]
```diff
@@ -0,0 +1,138 @@
...
+
+            drawerMenu.append(button)
+        }
+        return drawerMenu
+    }
+    
+    private lazy var menuStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: drawerMakeMenuButtons(drawerMenuItems))
+        stackView.axis = .vertical
+        stackView.translatesAutoresizingMaskIntoConstraints = false
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: you don't need to care about autoresizing mask. SnapKit is handling this for you

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 22]
```diff
@@ -0,0 +1,138 @@
...
+        super.init(coordinator: coordinator)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        setupView()
+    }
+    
+    private lazy var drawerMenuItems: [String] = [
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: make them into separated buttons, then compare sender directly to the button object in `func handleMenuTap()`

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 23]
```diff
@@ -0,0 +1,138 @@
...
+
+protocol DrawerCoordinator: BaseCoordinator {
+    func closeMenu()
+    func logout()
+    func deleteHistory()
+    func showAbout()
+}
+
+class DrawerViewController: BaseViewController {
+    private unowned let coordinator: DrawerCoordinator
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: // MARK: Instance part

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 24]
```diff
@@ -0,0 +1,138 @@
...
+    
+    init(coordinator: DrawerCoordinator) {
+        self.coordinator = coordinator
+        
+        super.init(coordinator: coordinator)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: // MARK: Lifecycle

File: zScanner/Views/Buttons/DrawerMenuButton.swift [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+//  DrawerMenuButton.swift
+//  zScanner
+//
+//  Created by Martin Georgiu on 11/08/2019.
+//  Copyright © 2019 Institut klinické a experimentální medicíny. All rights reserved.
+//
+
+import UIKit
+
+// MARK: -
```
Comment (skorepak) [inline zScanner/Views/Buttons/DrawerMenuButton.swift]: you can remove this MARK, I'm putting it to separate protocol definition, if there is any

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 26]
```diff
@@ -0,0 +1,142 @@
...
+            coordinator.logout()
+            
+        case deleteHistoryButton:
+            coordinator.deleteHistory()
+            
+        case aboutButton:
+            coordinator.showAbout()
+            
+        default:
+            print("bug")
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: Remove the print, you can use `assertionFailure()`, which will "crash" the app, but only in debug mode. Kind of forced breakpoint. You can also add message to it.

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 27]
```diff
@@ -0,0 +1,142 @@
...
+        drawerTopLabel.snp.makeConstraints { make in
+            make.top.equalTo(drawerLogo.snp.bottom).offset(20)
+            make.left.equalTo(topView).offset(20)
+            make.right.equalTo(topView).offset(20)
+            make.bottom.equalTo(topView).offset(-20)
+        }
+        
+        menuStackView.snp.makeConstraints { make in
+            make.top.equalTo(topView.snp.bottom).offset(15)
+            make.left.equalTo(view).offset(15)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: make.left.right.equalToSuperview.inset(15)

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 28]
```diff
@@ -0,0 +1,142 @@
...
+    private lazy var aboutButton: DrawerMenuButton = {
+        var button = DrawerMenuButton()
+        button.setTitle("drawer.aboutApp.title".localized, for: .normal)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(handleMenuTap(_:)), for: .touchUpInside)
+        return button
+    }()
+    
+    private lazy var menuStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [logoutButton, deleteHistoryButton, aboutButton])
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: move this line to `setupView()`

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 29]
```diff
@@ -0,0 +1,142 @@
...
+        button.setTitle("drawer.deleteHistory.title".localized, for: .normal)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.addTarget(self, action: #selector(handleMenuTap(_:)), for: .touchUpInside)
+        return button
+    }()
+    
+    private lazy var aboutButton: DrawerMenuButton = {
+        var button = DrawerMenuButton()
+        button.setTitle("drawer.aboutApp.title".localized, for: .normal)
+        button.translatesAutoresizingMaskIntoConstraints = false
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: remove this line from all the buttons :)

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 30]
```diff
@@ -0,0 +1,142 @@
...
+    private lazy var drawerLogo: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "menuLogo")
+        imageView.contentMode = UIView.ContentMode.scaleAspectFit
+        return imageView
+    }()
+    
+    private lazy var drawerTopLabel: UILabel = {
+        let label = UILabel()
+        label.text = "zScanner"
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: "drawer.header.title".localized

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 31]
```diff
@@ -0,0 +1,142 @@
...
+        label.textColor = .white
+        label.font = label.font.withSize(25)
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: there is UIFont extension. Use `label.font = .headline`

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 32]
```diff
@@ -0,0 +1,142 @@
...
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        setupView()
+    }
+    
+    private lazy var drawerLogo: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "menuLogo")
+        imageView.contentMode = UIView.ContentMode.scaleAspectFit
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: you can remove UIView.ContentMode and leave just `. scaleAspectFit`

File: zScanner/ViewControllers/Drawer/DrawerViewController.swift [hunk 33]
```diff
@@ -0,0 +1,142 @@
...
+        return imageView
+    }()
+    
+    private lazy var drawerTopLabel: UILabel = {
```
Comment (skorepak) [inline zScanner/ViewControllers/Drawer/DrawerViewController.swift]: // MARK: Helpers

File: zScanner/ViewControllers/About/AboutViewController.swift [hunk 34]
```diff
@@ -0,0 +1,85 @@
...
+    private lazy var stackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [drawerLogo,aboutHeader, aboutParagraph, aboutCopyright])
+        stackView.axis = .vertical
+        stackView.alignment = .center
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.spacing = 15
+        return stackView
+    }()
+    
+    func setupView() {
```
Comment (skorepak) [inline zScanner/ViewControllers/About/AboutViewController.swift]: make it private

File: zScanner/ViewControllers/About/AboutViewController.swift [hunk 35]
```diff
@@ -0,0 +1,85 @@
...
+        navigationItem.title = "drawer.aboutApp.title".localized
+      
+        self.view.addSubview(stackView)
+        
+        // MARK: - Constraints
```
Comment (skorepak) [inline zScanner/ViewControllers/About/AboutViewController.swift]: Why mark here? make it just comment

File: zScanner/ViewControllers/About/AboutViewController.swift [hunk 36]
```diff
@@ -0,0 +1,85 @@
...
+        super.init(coordinator: coordinator)
+    }
+    
+    override func loadView() {
+        super.loadView()
+        
+        setupView()
+    }
+    
+    private lazy var drawerLogo: UIImageView = {
```
Comment (skorepak) [inline zScanner/ViewControllers/About/AboutViewController.swift]: // MARK: Helpers

File: zScanner/ViewControllers/About/AboutViewController.swift [hunk 37]
```diff
@@ -0,0 +1,85 @@
...
+    
+    // MARK: - Instance part
+    private unowned let coordinator: MenuCoordinator
+    
+    init(coordinator: MenuCoordinator) {
+        self.coordinator = coordinator
+        super.init(coordinator: coordinator)
+    }
+    
+    override func loadView() {
```
Comment (skorepak) [inline zScanner/ViewControllers/About/AboutViewController.swift]: // MARK: Lifecycle

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 38]
```diff
@@ -0,0 +1,96 @@
...
+            self.blackView.alpha = 0
+            self.drawerViewController.view.frame.origin = CGPoint(x: -self.drawerViewController.view.frame.width, y: 0)
+        }
+    }
+    
+    @objc private func handleDismiss(){
+        closeMenu()
+    }
+    
+    func showAbout() {
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: This function should be in the DrawerCoordinator extension

Comment (skorepak) [reply zScanner/Coordinators/MenuCoordinator.swift]: More like rename it into `private showAboutScreen` and call this function from the extension. We should prevent "heavy"functions in the extensions

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 39]
```diff
@@ -0,0 +1,96 @@
...
+    private lazy var menuWidth: CGFloat = {
+        let width = window.frame.width * 2 / 3
+        return width
+    }()
+
+    func begin() {
+        installDrawerSreen()
+    }
+    
+    private func installDrawerSreen() {
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: group this function with other private functions at the bottom of the scope and add marks `Interface` and `Private`

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 40]
```diff
@@ -0,0 +1,96 @@
...
+
+class MenuCoordinator: Coordinator {
+    // MARK: Instance part
+    unowned private let flowDelegate: MenuFlowDelegate
+
+    init(flowDelegate: MenuFlowDelegate, window: UIWindow, navigationController: UINavigationController? = nil) {
+        self.flowDelegate = flowDelegate
+
+        super.init(window: window, navigationController: navigationController)
+
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: remove empty line 😄

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 41]
```diff
@@ -0,0 +1,96 @@
...
+import UIKit
+
+protocol MenuFlowDelegate: FlowDelegate {
+    func logout()
+    func deleteHistory()
+}
+
+class MenuCoordinator: Coordinator {
+    // MARK: Instance part
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: add empty line 😄

File: zScanner/Coordinators/MenuCoordinator.swift [hunk 42]
```diff
@@ -0,0 +1,96 @@
...
+//
+
+import UIKit
+
+protocol MenuFlowDelegate: FlowDelegate {
+    func logout()
+    func deleteHistory()
+}
+
+class MenuCoordinator: Coordinator {
```
Comment (skorepak) [inline zScanner/Coordinators/MenuCoordinator.swift]: // MARK: -

File: zScanner/Coordinators/DocumentsCoordinator.swift [hunk 43]
```diff
@@ -69,3 +85,15 @@ extension DocumentsCoordinator: NewDocumentFlowDelegate {
...
+
+// MARK: - MenuFlowDelegate implementation
+extension DocumentsCoordinator: MenuFlowDelegate {
+    func deleteHistory() {
+        database.deleteAll(of: DocumentDatabaseModel.self)
+        database.deleteAll(of: FolderDatabaseModel.self)
+    }
+    
+    func logout() {
+        flowDelegate.logout()
```
Comment (skorepak) [inline zScanner/Coordinators/DocumentsCoordinator.swift]: we need to delete the history on login and then remove this coordinator from the memory

```
deleteHistory()
flowDelegate.logout()
flowDelegate.coordinatorDidFinish(self)
```

File: zScanner/Coordinators/DocumentsCoordinator.swift [hunk 44]
```diff
@@ -28,13 +30,24 @@ class DocumentsCoordinator: Coordinator {
...
         showDocumentsListScreen()
+        setupMenu()
     }
     
     // MARK: Navigation methods
     private func showDocumentsListScreen() {
         let viewModel = DocumentsListViewModel(database: database, ikemNetworkManager: networkManager)
         let viewController = DocumentsListViewController(viewModel: viewModel, coordinator: self)
         push(viewController)
+
```
Comment (skorepak) [inline zScanner/Coordinators/DocumentsCoordinator.swift]: remove empty line

File: zScanner/Coordinators/AppCoordinator.swift [hunk 45]
```diff
@@ -18,8 +18,8 @@ class AppCoordinator: Coordinator {
     
     // MARK: Inteface
     func begin() {
-        startLoginCoordinator()
-//        startDocumentsCoordinator()
```
Comment (skorepak) [inline zScanner/Coordinators/AppCoordinator.swift]: Remove this change. It's for debug and it's not in "scope"of the task you are doing :) Try to prevent any unnecessary changes.

File: zScanner/ViewControllers/About/AboutViewController.swift [hunk 46]
```diff
@@ -0,0 +1,90 @@
...
+        imageView.contentMode = .scaleAspectFit
+        imageView.backgroundColor = .primary
+        return imageView
+    }()
+    
+    private lazy var aboutHeader: UILabel = {
+        let label = UILabel()
+        label.text = "about.header.title".localized
+        label.textColor = .primary
+        label.font = label.font.withSize(30)
```
Comment (skorepak) [inline zScanner/ViewControllers/About/AboutViewController.swift]: use font `.header `

File: zScanner/ViewControllers/About/AboutViewController.swift [hunk 47]
```diff
@@ -0,0 +1,90 @@
...
+        label.textColor = .primary
+        return label
+    }()
+    
+    private lazy var aboutCopyright: UILabel = {
+        let label = UILabel()
+        label.text = "about.copyright.title".localized
+        label.textAlignment = .center
+        label.textColor = .primary
+        label.font = label.font.withSize(14)
```
Comment (skorepak) [inline zScanner/ViewControllers/About/AboutViewController.swift]: use `.footnote`

File: zScanner/ViewControllers/About/AboutViewController.swift [hunk 48]
```diff
@@ -0,0 +1,90 @@
...
+        stackView.axis = .vertical
+        stackView.alignment = .center
+        stackView.spacing = 15
+        return stackView
+    }()
+    
+    private func setupView() {
+        navigationItem.title = "drawer.aboutApp.title".localized
+      
+        stackView.addArrangedSubview(drawerLogo)
```
Comment (skorepak) [inline zScanner/ViewControllers/About/AboutViewController.swift]: oh, I didn't know there is no function to add an array at once. 

If you want you can do `[drawerLogo, aboutHeader, ...].forEach({ stackView. addArrangedSubview($0) })` up to you, what do you like more :)

File: zScanner/ViewControllers/About/AboutViewController.swift (hunk 46 above)
Comment (killalad) [reply zScanner/ViewControllers/About/AboutViewController.swift]: do you mean `.headline`?

Comment (skorepak) [reply zScanner/ViewControllers/About/AboutViewController.swift]: yep, sorry
